_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,667 @@
+
+This command created a config file named `.okta.env`. It will look something like the following. It helpfully lists the values you will need in the next step.
+
+```bas
+export SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_OIDC_ISSUER_URI="https://dev-123456/oauth2/default"
+export SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_ID="2989u928u383..."
+export SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET="09328uu098u4..."
+```
+
+Both the Docker Compose project and the Kubernetes project use Spring Cloud Config to centralize configuration. The JHipster registry service makes these config values available to all of the other services in the cluster.
+
+Open `docker-compose/central-server-config/application.yml` and add the following to the end, filling in the `issuer-uri`, `client-id`, and `client-secret` taken from the `.okta.env` file. This is the Spring Cloud Config file for the Docker Compose project.
+
+`docker-compose/central-server-config/application.yml`
+
+```yml
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          oidc:
+            issuer-uri: https://<your-okta-domain>/oauth2/default
+        registration:
+          oidc:
+            client-id: <client-id>
+            client-secret: <client-secret>
+```
+
+#### Build the Docker Images and Run the App with Docker Compose
+
+You're all set to run the app locally using Docker and Docker Compose. You need to build the docker image for each of the projects: `gateway`, `store`, and `blog`.
+
+ In the three different app directories, run the following Gradle command.
+
+```
+./gradlew -Pprod bootJar jibDockerBuild
+```
+
+Navigate to the `docker-compose` directory and run the app. You can use the `-d` param to run it as a daemon but for the moment I like seeing the logs. You're just running this in Docker Compose as a warm-up for the Azure deployment anyway.
+
+```bash
+docker-compose up
+```
+
+Give that a minute or two to finish running all the services. 
+
+Open the gateway service: http://localhost:8080/
+
+Go to **Account** and **Sign in**. You should be directed to the Okta sign-in form. 
+
+{% img blog/jhipster-k8s-azure/signing-in.png alt:"Okta Login" width:"500" %}{: .center-image }
+
+
+You should be able to authenticate with your Okta credentials.
+
+{% img blog/jhipster-k8s-azure/authenticated.png alt:"Authenticated" width:"800" %}{: .center-image }
+
+Make sure all parts of the application are working. First, test the store (which is using the Cosmos DB Mongo database) by going to **Entities** and **Product**. Make sure you can add a new product and see it in the list of products. Next, create a blog (**Entities** and **Blog**).
+
+If that works, you can take a look at the **Administration** menu. There's a lot of helpful info there. 
+
+You can also check the JHipster Registry: http://localhost:8761/
+
+{% img blog/jhipster-k8s-azure/registry.png alt:"Authenticated" width:"800" %}{: .center-image }
```

Comment:
```suggestion
{% img blog/jhipster-k8s-azure/registry.png alt:"JHipster Registry" width:"800" %}{: .center-image }
```

Reply:
This screenshot is a little hard to read. I just tried taking my own and resizing to 1600w. It doesn't look much better, so maybe it's good enough? 

<img width="800" alt="jhipster-registry" src="https://user-images.githubusercontent.com/17892/165599117-43b61fbf-db61-4157-a3e5-e7a39bab1b6c.png">